### ISSUE TITLE
Trying to find % of total number of device offline at every 15 minutes

### ISSUE THREAD

**Issue by hinguabhishek** [confluentinc/ksql#1374]
I am trying to do aggregation on live sensor data ingest into kafka topic at every 15 mins. i want to identify file % of inactive device at every 15 minutes interval so for this use case I have return below KSQL Query.

`./bin/kafka-avro-console-producer --broker-list localhost:9092 --topic  DEVICE --property value.schema='{"type":"record","name":"myrecord","fields":[{"name":"RecordDate","type":"long"},{"name":"DeviceId","type":"string"},{"name":"DeviceType","type":"string"},{"name":"DeviceStatus","type":"string"},{"name":"VehicleId","type":"int"},{"name":"isDeviceOnline","type":"int"}]}' 
`

```
CREATE STREAM  DEVICE_STREAM WITH \ 
(KAFKA_TOPIC='DEVICE',TIMESTAMP='RecordDate',KEY='DEVICEID',VALUE_FORMAT='AVRO');

CREATE TABLE  ACTIVE_DEVICE_AT_15MIN AS \ 
SELECT RECORDDATE,COUNT(*) AS ACTIV_COUNT \
FROM  DEVICE_STREAM WINDOW TUMBLING(SIZE 15 MINUTES) \
WHERE DEVICESTATUS='1' GROUP BY RECORDDATE; /*Active Device*/

CREATE TABLE  TOTAL_DEVICE_AT_15MIN AS \
SELECT RECORDDATE,COUNT(*) AS TOTAL_DEVICE \
FROM  DEVICE_STREAM \
WINDOW TUMBLING(SIZE 15 MINUTES) \
GROUP BY RECORDDATE; /*Total Device*/

CREATE TABLE  INACTIVE_DEVICE_AT_15MIN AS \
SELECT RECORDDATE,COUNT(*) AS INACTIVE_COUNT \
FROM  DEVICE_STREAM \
WINDOW TUMBLING(SIZE 15 MINUTES) \
WHERE DEVICESTATUS!='1' GROUP BY RECORDDATE; /*Active Device*/

CREATE STREAM  INACTIVE_DEVICE_INFO AS \
SELECT  INACTIVE_DEVICE_AT_15MIN.RECORDDATE, \
    INACTIVE_DEVICE_AT_15MIN.INACTIVE_COUNT, \
    TOTAL_DEVICE_AT_15MIN.TOTAL_DEVICE  \
FROM  TOTAL_DEVICE_AT_15MIN \
  LEFT JOIN  INACTIVE_DEVICE_AT_15MIN \
  ON  INACTIVE_DEVICE_AT_15MIN.RECORDDATE =  TOTAL_DEVICE_AT_15MIN.RECORDDATE; /*get  inactive and total device count at every 15 mins to calculate % device offline at every 15 min.
```

When I try to run below join above join statement it reurn me error

```
Cannot RUN execution plan for this statement, CreateStreamAsSelect{name=INACTIVE_DEVICE_INFO, query=Query{queryBody=QuerySpecification{select=Select{distinct=false, selectItems=[INACTIVE_DEVICE_AT_15MIN.RECORDDATE INACTIVE_DEVICE_AT_15MIN_RECORDDATE, INACTIVE_DEVICE_AT_15MIN.INACTIVE_COUNT INACTIVE_COUNT, TOTAL_DEVICE_AT_15MIN.TOTAL_DEVICE TOTAL_DEVICE]}, from=Join{type=LEFT, left=AliasedRelation{relation=TOTAL_DEVICE_AT_15MIN, alias=TOTAL_DEVICE_AT_15MIN}, right=AliasedRelation{relation=INACTIVE_DEVICE_AT_15MIN, alias=INACTIVE_DEVICE_AT_15MIN}, criteria=Optional[JoinOn{(INACTIVE_DEVICE_AT_15MIN.RECORDDATE = TOTAL_DEVICE_AT_15MIN.RECORDDATE)}]}, =null, where=null, groupBy=Optional.empty, having=null, limit=null}}, notExists=false, properties={}}
Caused by: null
```

**Comment by rmoff** [comment]
Related: #1422 

You're trying to do a TABLE-TABLE join, which is not currently supported, but is coming (#760). The error you got is addressed in #1133.

**Comment by rmoff** [comment]
BTW I notice that you're grouping by `RECORDDATE`, as well as using it in the `TIMESTAMP` value for the `CREATE STREAM`. If this is just a date (no time) then it won't be valid as a `TIMESTAMP`. If it is including time then you're going to end up with multiple buckets in your `GROUP BY`.
